modin/experimental/core/execution/native/implementations/omnisci_on_native/partitioning/partition.py

```diff
@@ -62,6 +65,12 @@ def __init__(
         self.arrow_table = arrow_table
         self._length_cache = length
         self._width_cache = width
+        self._server = OmnisciServer
+
+    def __del__(self):
+        """Deallocate OmniSci resources related to the partition."""
+        if self.frame_id is not None:
+            self._server.executeDDL(f"DROP TABLE {self.frame_id};")
```

Comment:
```suggestion
            self._server.executeDDL(f"DROP TABLE IF EXISTS {self.frame_id};")
```
Even though we check for `self.frame_id is not None` above, shouldn't we prevent the case when a user directly uses this Partition class to set `frame_id` to some value but doesn't yet import arrow table to OmniSci?

Reply:
Direct partition manipulations are always at the user's own risk. I'd say it is an error to have `frame_id` set to an inexistent table. It's better to get an error message in such a case than to silently ignore that fact.

Reply:
Makes sense